src/bootstrap/builder.rs

```diff
@@ -1139,10 +1139,18 @@ impl<'a> Builder<'a> {
         // itself, we skip it by default since we know it's safe to do so in that case.
         // See https://github.com/rust-lang/rust/issues/79361 for more info on this flag.
         if target.contains("apple") {
-            if self.config.rust_run_dsymutil {
-                rustflags.arg("-Zrun-dsymutil=yes");
+            if stage == 0 {
+                if self.config.rust_run_dsymutil {
+                    rustflags.arg("-Zrun-dsymutil=yes");
+                } else {
+                    rustflags.arg("-Zrun-dsymutil=no");
+                }
```

Comment:
FYI, it would be helpful next time to add `// cfg(bootstrap)` as a comment, this caused a failure that wasn't caught until testing on Mac: https://github.com/rust-lang/rust/pull/82076#issuecomment-780253845